oeps/architectural-decisions/oep-0058-arch-translations-management.rst

```diff
@@ -0,0 +1,263 @@
+.. _this pull request: https://github.com/openedx/edx-platform/pull/30567
+
+Rationale for migrating to the `Transifex GitHub App`_
+******************************************************
+
+* This is an upgrade of a system we use regularly, but do not want to have to maintain
+  regularly.
+* Upgrading from a bot (machine user) to an app/workflow is recommended by GitHub and
+  makes the translation process more open source. 
+* The `Transifex GitHub App`_ is developed and supported by Transifex
+* The `Transifex GitHub App`_ is very simple to configure and has many options. We can
+  set Transifex Projects to automatically upload/download translation files from a
+  repository once the translations are reviewed and accepted.
+* By using an app that is maintained by Transifex the organization, we reduce the
+  maintenance burden and are more future proof of changes they might make since they
+  maintain both the API and the `Transifex GitHub App`_.
+
+Rationale for consolidating translations files centrally
+********************************************************
+
+* Transifex only allows a one-to-one relationship between repositories and Transifex
+  Projects. Organizing all of the translation files into one repository and one Transifex
+  Project has a lower labor cost: projects are managed separately so we end up spending
+  less time tracking translation progress, and debugging translation issues when all
+  translation files are put in the same place. By decreasing the number of projects we
+  need to maintain, we can add more content like the MFE translations.
+* A repository that only contains text/binary files, and uses branches to separate
+  translations related to Open edX releases can make all interactions with translations
+  very quick and simple due to the ability to clone and sparse-checkout the branch of a
+  specific release and the directory (repository name) with translation files.
+
+Proposed Implementation
+***********************
+
+Move Translation Files to a New Repo
+====================================
+
+Translation files (of types .mo, .po, and .json) currently exist amongst the code and
+documentation they translate. We will move these translation files from being amongst the
+code/documentation to their own repo. For example, a translation file for the openedx
+repository `edx-platform`_ located at
+``edx-platform/conf/locale/en/LC_MESSAGES/django.po`` would be moved to the new
+repository with the name openedx-translations_ located at
+``openedx-translation/edx-platform/conf/locale/en/LC_MESSAGES/django.po``. For easier
+reintegration, translation files will be kept in the same directory structure as the
+code/documentation they translate.
+
+Repositories that generate translation files will have their translation files generated
+and committed via a pull request to the openedx-translation repository via a GitHub
+workflow. Once the translation files from edx-platform and other repositories are moved
```

Comment:
What will this GitHub workflow be called?

Reply:
https://github.com/openedx/openedx-translations/blob/main/.github/workflows/extract-translation-source-files.yml

Reply:
Thank you! Would it be worth adding that link into the OEP?

Reply:
@pshiu from [OEP-1](https://open-edx-proposals.readthedocs.io/en/latest/processes/oep-0001.html):
> OEPs are not used to dictate small decisions made in every day feature work. See [OEP-19](https://open-edx-proposals.readthedocs.io/en/latest/best-practices/oep-0019-bp-developer-documentation.html?highlight=documentation#agile-documentation-types) for more detail.

I think the workflow name is a small decision. I wouldn't want to lock Carlos et al into that specific workflow name if it's not critical to this OEP's deisgn. Do you see any harm in leaving it out?

Reply:
Not at all! Thought it might be a nice add if it was already decided for reference.

Reply:
Since OEP-58's next target status is "Provisional", once it is back "Under Review" with the target status of "Accepted", we can update the details of OEP-58 to reflect the specifics of implementation.